### PR TITLE
Implement warning when leaving the page and auto-saving when returning

### PR DESCRIPTION
**Pull request by Minh13**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] ~~Server: I added multiple integration tests (Spring) related to the features~~
- [x] ~~Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)~~
- [x] ~~Server: I implemented the changes with a good performance and prevented too many database calls~~
- [x] ~~Server: I documented the Java code using JavaDoc style.~~
- [x] ~~Client: I added multiple integration tests (Jest) related to the features~~
- [x] ~~Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~~
- [x] ~~Client: I documented the TypeScript code using JSDoc style.~~
- [x] ~~Client: I added multiple screenshots/screencasts of my UI changes~~
- [x] ~~Client: I translated all the newly inserted strings into German and English~~

### Motivation and Context

[https://github.com/ls1intum/Artemis/issues/498](url)

### Description
When exiting the text editor of text exercises, the given answer would not be saved automatically and changes might get lost. This is solved by saving when the component is destroyed and triggering a browser based warning when the page is refreshed/browser is closed.
### Steps for Testing

1. Log in to Artemis
2. Navigate to a Course
3. Navigate to an exercise of type 'text'
4. Give a text answer and exit w/o saving

### CODE REVIEW

**Comment by sascha11110** [comment]
Just for the workflow: check all points of the checklist, even if you have crossed them out :-) Otherwise it looks from the outside as if not everything is ready yet.

File: src/main/webapp/app/text-editor/text-editor.component.ts [hunk 1]
```diff
@@ -111,6 +113,22 @@ export class TextEditorComponent implements OnInit {
...
     }
 
+    ngOnDestroy() {
+        if (this.canDeactivate() && this.textExercise.id) {
+            if (!this.submission) {
+                this.submission = new TextSubmission();
+            }
+            this.submission.submitted = false;
+            this.submission.text = this.answer;
+            this.textSubmissionService[this.submission.id ? 'update' : 'create'](this.submission, this.textExercise.id).subscribe(response => {
```
Comment (thilo-behnke) [inline src/main/webapp/app/text-editor/text-editor.component.ts]: If somebody would e.g. rename `textSubmissionService.update`, this place would not get caught by the refactoring. It's slick but we should use the longer direct method reference instead.

File: src/main/webapp/app/text-editor/text-editor.component.ts [hunk 2]
```diff
@@ -111,6 +113,22 @@ export class TextEditorComponent implements OnInit {
         );
     }
 
+    ngOnDestroy() {
+        if (this.canDeactivate() && this.textExercise.id) {
+            if (!this.submission) {
+                this.submission = new TextSubmission();
+            }
+            this.submission.submitted = false;
```
Comment (thilo-behnke) [inline src/main/webapp/app/text-editor/text-editor.component.ts]: There is no point in setting any of the component properties in ngOnDestroy as they will be deleted anyway. Just to make the intention clearer, I would create a throwaway `let` and don't assign anything in the subscribe.

Comment (Minh13) [reply src/main/webapp/app/text-editor/text-editor.component.ts]: Fixed in https://github.com/ls1intum/Artemis/pull/971/commits/67e0295ce1158503d63d21d9bf6a3221531fbe6a

File: src/main/webapp/app/text-editor/text-editor.component.ts (hunk 1 above)
Comment (Minh13) [reply src/main/webapp/app/text-editor/text-editor.component.ts]: Fixed in https://github.com/ls1intum/Artemis/pull/971/commits/d1898947418783f542c86a1ab3adc1cd32649164

**Pull request merged**